### ISSUE TITLE
Document supported Markdown flavour

### ISSUE THREAD

**Issue by davidmehren** [hedgedoc/hedgedoc#928]
We should document what flavor of Markdown we support.

We use [markdown-it](https://github.com/markdown-it/markdown-it), which supports CommonMark plus tables and strikethrough from [GitHub Flavored Markdown](https://github.github.com/gfm).
The following plugins are added (excerpt from 1.x `package.json`):
```
"markdown-it-abbr": "^1.0.4",
"markdown-it-container": "^3.0.0",
"markdown-it-deflist": "^2.0.1",
"markdown-it-emoji": "^2.0.0",
"markdown-it-footnote": "^3.0.1",
"markdown-it-imsize": "^2.0.1",
"markdown-it-ins": "^3.0.0",
"markdown-it-mark": "^3.0.0",
"markdown-it-mathjax": "^2.0.0",
"markdown-it-regexp": "^0.4.0",
"markdown-it-sub": "^1.0.0",
"markdown-it-sup": "^1.0.0",
```
This might bring us closer to full GFM-compatibility, but also has many more extensions, so we probably end up with a weird superset of CommonMark, that might be GFM but not really.

I would like to have a table in the docs that list the various features of CommonMark, GFM and HedgeDocMark™, so the compatibility is made clear.

**Comment by DerMolly** [comment]
Started work on such a list [here](https://demo.hedgedoc.org/rz_x0gn5Sxmfb685UuuTXw)

**Comment by SISheogorath** [comment]
While I fully agree with everything stated above, the name definitely should be "HedgeDoc-Flavoured-Markdown" -> HFM. 😉

**Comment by zeigerpuppy** [comment]
@DerMolly that table is really useful, thanks!
It would also be good to have some further development documentation on how to implement/include new plugins.  For instance, I can see some utility in implementing https://www.npmjs.com/package/markdown-it-contained-image to allow having images in divs which specify position.

**Comment by DerMolly** [comment]
Yes this is planned, but currently HedgeDoc 2.0 is in a heavy development phase and not everything in the new frontend is totally built in stone. Also people currently like to invest their time in building new features and not writing documentation. 
If you want to write this part of the developer documentation, I could give some hints in the chat…
